src/TNode.h

```diff
@@ -0,0 +1,54 @@
+//
+// Created by shvmo on 29/04/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_TNODE_H
+#define DATASTRUCTURESWET1_TNODE_H
+
+
+#include <memory>
+
+using std::shared_ptr;
+
+template<class Key, class Info>
```

Comment:
if we called it key then we should probably rename `Info` to `Value`
so we'll have key-value terminology 😃

Reply:
fiiixed